src/Apps/Order/Routes/Offer/index.tsx

```diff
@@ -54,6 +54,19 @@ export const OfferRoute: FC<OfferRouteProps> = ({
   })
   const [offerValue, setOfferValue] = useState(0)
 
+  useEffect(() => {
+    if (order.myLastOffer?.note) {
+      if (!order.myLastOffer.note.startsWith("I sent an offer for")) {
+        setOfferNoteValue({
+          exceedsCharacterLimit: false,
+          value: order.myLastOffer.note,
+        })
+      }
+    }
+    // need this to run only once
```

Comment:
Since this value only needs to be set once to initialize the state variable, can we put this logic into a function called `initialNoteValue(note)` and use it when we call `useState` instead?

```js
const [offerNoteValue, setOfferNoteValue] = useState({
  exceedsCharacterLimit: false,
  value: initialNoteValue(order.myLastOffer?.note),
})
```

Reply:
Something similar is happening in a `useEffect` hook in the PriceOptions component, and I think that logic could also be simplified by performing that initialization in the `useState` call rather than an effect.